src/middlewares/user.ts

```diff
@@ -365,155 +356,3 @@ export const checkUserHasSelectedAnOrganizationMiddleware = (
-          userOrganisations.find(
-            ({ id, needs_official_contact_email_verification }) =>
-              needs_official_contact_email_verification &&
-              id === selectedOrganizationId,
-          );
-      } else {
-        organizationThatNeedsOfficialContactEmailVerification =
-          userOrganisations.find(
-            ({ needs_official_contact_email_verification }) =>
-              needs_official_contact_email_verification,
-          );
-      }
-
-      if (!isEmpty(organizationThatNeedsOfficialContactEmailVerification)) {
-        return res.redirect(
-          `/users/official-contact-email-verification/${organizationThatNeedsOfficialContactEmailVerification.id}`,
-        );
-      }
-
-      return next();
-    } catch (error) {
-      next(error);
-    }
-  });
-
-export const checkUserHasBeenAuthenticatedByPeersMiddleware = (
-  req: Request,
-  res: Response,
-  next: NextFunction,
-) =>
-  checkUserHasNoPendingOfficialContactEmailVerificationMiddleware(
-    req,
-    res,
-    async (error) => {
-      try {
-        if (error) return next(error);
-
-        const { id: user_id, email } = getUserFromAuthenticatedSession(req);
-
-        const userOrganisations = await getOrganizationsByUserId(user_id);
-
-        let organizationThatNeedsAuthenticationByPeers;
-        if (req.session.mustReturnOneOrganizationInPayload) {
-          const selectedOrganizationId =
-            await getSelectedOrganizationId(user_id);
-
-          organizationThatNeedsAuthenticationByPeers = userOrganisations.find(
-            ({ id, authentication_by_peers_type }) =>
-              !authentication_by_peers_type && id === selectedOrganizationId,
-          );
-        } else {
-          organizationThatNeedsAuthenticationByPeers = userOrganisations.find(
-            ({ authentication_by_peers_type }) => !authentication_by_peers_type,
-          );
-        }
-
-        if (!isEmpty(organizationThatNeedsAuthenticationByPeers)) {
-          const organization_id = organizationThatNeedsAuthenticationByPeers.id;
-          const internalActiveUsers =
-            await getInternalActiveUsers(organization_id);
-          const otherInternalUsers = internalActiveUsers.filter(
-            ({ email: e }) => e !== email,
-          );
-
-          if (PAIR_AUTHENTICATION_WHITELIST.includes(getEmailDomain(email))) {
-            await markAsWhitelisted({ user_id, organization_id });
-          } else if (usesAGouvFrDomain(email)) {
-            await markAsGouvFrDomain({ user_id, organization_id });
-          } else if (otherInternalUsers.length > 0) {
-            return res.redirect(`/users/choose-sponsor/${organization_id}`);
-          } else {
-            await notifyAllMembers({ user_id, organization_id });
-          }
-        }
-
-        return next();
-      } catch (error) {
-        next(error);
-      }
-    },
-  );
-
-export const checkUserHasBeenGreetedForJoiningOrganizationMiddleware = (
```

Comment:
keep this!

Reply:
Why do we need to keep the checkUserHasBeenGreetedForJoiningOrganisationMiddleware function if it only uses the checkUserHasBeenAuthenticatedByPeersMiddleware function?
Maybe it's a naming problem

Reply:
@rebeccadumazert We discussed this on Friday: it's a more convenient and meaningful alias since this middleware is among the most frequently used.

Reply:
@rdubigny yes, it's a naming problem :
```
export const checkUserHasBeenGreetedForJoiningOrganizationMiddleware = (
  req: Request,
  res: Response,
  next: NextFunction,
) =>
  checkUserHasBeenAuthenticatedByPeersMiddleware(req, res, async (error) => {
  });
```

We need to rename the function `checkUserHasBeenAuthenticatedByPeersMiddleware`, it's a wrong name.

Reply:
that's what confused me 🫠😶‍🌫️

Reply:
seen IRL